electrum/wallet.py

```diff
@@ -1107,19 +1099,12 @@ def _is_onchain_invoice_paid(self, invoice: Invoice) -> Tuple[bool, Optional[int
 
     def is_onchain_invoice_paid(self, invoice: Invoice) -> Tuple[bool, Optional[int]]:
         is_paid, conf_needed, relevant_txs = self._is_onchain_invoice_paid(invoice)
+        if is_paid:
+            key = self.get_key_for_outgoing_invoice(invoice)
+            for txid in relevant_txs:
+                self._invoices_from_txid_map[txid].add(key)
```

Comment:
I don't think it's nice to add a side-effect to this method.
I find it difficult to find the code path where this would get called that we actually rely on, for the side-effect. Is it somewhere in the GUI where the invoice status will get checked? We should not rely on that...

Reply:
Indeed. I guess one could try to detect invoice payment in `on_event_adb_added_tx`, similar to `_update_request_statuses_touched_by_tx`

Reply:
sounds reasonable